### PR TITLE
Recording Proxy test coverage

### LINKED ISSUE #1496: Recording Proxy Tests
For the recording proxy to be successful, we need to test and make sure the following work:

- [ ] Sessions with OpenSSH clients terminated, forwarded, and recorded to a local cluster.
- [ ] Sessions with OpenSSH clients terminated, forwarded, and recorded to a local cluster.
- [ ] Sessions with OpenSSH clients terminated, forwarded, and recorded to a remote cluster.
- [ ] Sessions with OpenSSH clients terminated, forwarded, and recorded to a remote cluster.
- [ ] `scp` use is emitted to the audit log.
- [ ] Agent can be forwarded to the proxy and the destination node.
- [ ] Support for remote shell, exec (like scp), and all subsystems (like sftp).
- [ ] Passing of Teleport specific ciphers.
- [ ] RBAC works to the extent possible (no labels).
- [ ] Port forwarding.
- [ ] Correct terminal type is forwarded.

### PR DESCRIPTION
**Pull request by russjones**
**Purpose**

Added integration test coverage and fixes for minor issues discovered during tests.

**Implementation**

* Integration test coverage for the forwarding server and recording proxy.
* Pass ciphers, kex and mac algorithms to the forwarding server.
* Made `forward_agent: true` a default for the admin role.
* Migrated older roles that don't contain port forwarding and agent forwarding to now contain these in role options.
* Fixed regression in `UserKeyAuth`.
* When a remote site dials, it allows the server to decide which auth server to connect to.

**Related Issues**

Fixes https://github.com/gravitational/teleport/issues/1496

### CODE REVIEW

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Review by kontsevoy — APPROVED** [review]

File: integration/integration_test.go [hunk 1]
```diff
@@ -957,6 +1148,279 @@ func (s *IntSuite) TestDiscovery(c *check.C) {
 	c.Assert(main.Stop(true), check.IsNil)
 }
 
+// TestExternalClient tests if we can connect to a node in a Teleport
+// cluster. Both normal and recording proxies are tested.
+func (s *IntSuite) TestExternalClient(c *check.C) {
+	// only run this test if we have access to the external ssh binary
+	_, err := exec.LookPath("ssh")
+	if err != nil {
+		return
```
Comment (klizhentas) [inline integration/integration_test.go]: use `c.Skip` with message, that will show up in stats and message will be displayed

File: lib/auth/init.go [hunk 2]
```diff
@@ -414,20 +414,32 @@ func migrateRoles(asrv *AuthServer) error {
 		return trace.Wrap(err)
 	}
 
-	// loop over all roles and only migrate RoleV2 -> RoleV3
+	// DEPRECATED: Remove this migration in Teleport 2.5.
```
Comment (klizhentas) [inline lib/auth/init.go]: I suggest to use tag: `DELETE IN: 2.5.0` as a first line and then second line should explain why.

**Review by klizhentas — APPROVED** [review]

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Pull request merged**